WordPress/src/main/java/org/wordpress/android/analytics/AnalyticsTrackerNosara.java

```diff
@@ -0,0 +1,385 @@
+                eventName = "support_opened_helpshift_screen";
+                break;
+            case LOGIN_FAILED:
+                eventName = "login_failed_login";
+                break;
+            case LOGIN_FAILED_TO_GUESS_XMLRPC:
+                eventName = "login_failed_to_guess_xmlrpc";
+                break;
+            default:
+                eventName = null;
+                break;
+        }
+
+
+        if (eventName == null) {
+            Log.w(LOGTAG, "There is NO match for the event " + stat.name() + "stat");
+            return;
+        }
+
+        if (shouldGenerateAnonID()) {
+            mAnonID = getNewAnonID();
+        }
+
+        final String user = mAnonID != null ? mAnonID : mWpcomUserName;
+        TracksClient.NosaraUserType userType = mAnonID != null ?
+                TracksClient.NosaraUserType.ANON :
+                TracksClient.NosaraUserType.WPCOM;
+
+        if (properties != null && properties.size() > 0) {
+            JSONObject propertiesToJSON = new JSONObject(properties);
+            mNosaraClient.track(EVENTS_PREFIX + eventName, propertiesToJSON, user, userType);
+        } else {
+            mNosaraClient.track(EVENTS_PREFIX + eventName, user, userType);
+        }
+    }
+
+    @Override
+    public void beginSession() {
+        if (mNosaraClient == null) {
+            return;
+        }
+
+        refreshMetadata();
+    }
+
+    private boolean shouldGenerateAnonID() {
+        return (mAnonID == null && mWpcomUserName == null);
+    }
+
+    private String getNewAnonID() {
+        String uuid = UUID.randomUUID().toString();
+        String[] uuidSplitted = uuid.split("-");
+        StringBuilder builder = new StringBuilder();
+        for (String currentPart : uuidSplitted) {
+            builder.append(currentPart);
+        }
+        uuid = builder.toString();
+        Log.d(LOGTAG, "anon UUID generato " + uuid);
+        return uuid;
+    }
+
+    @Override
+    public void endSession() {
+        if (mNosaraClient == null) {
+            return;
+        }
+        mNosaraClient.flush();
+    }
+
+    @Override
+    public void refreshMetadata() {
+        if (mNosaraClient == null) {
+            return;
+        }
+
+        boolean connected = WordPress.hasDotComToken(WordPress.getContext());
+        if (connected) {
+            SharedPreferences preferences = PreferenceManager.getDefaultSharedPreferences(WordPress.getContext());
+            String username = preferences.getString(WordPress.WPCOM_USERNAME_PREFERENCE, null);
+            mWpcomUserName = username;
+            // Re-unify the user
+            if (mAnonID != null) {
+                mNosaraClient.trackAliasUser(mWpcomUserName, mAnonID);
+                mAnonID = null;
+            }
+        } else {
+            // Not wpcom connected. Check if mAnon is already present
+            mWpcomUserName = null;
+            if (shouldGenerateAnonID()) {
+                mAnonID = getNewAnonID();
```

Comment:
<blockquote>
Just to make sure: we're generating a new ID each time the app restarts (when mAnonID state is lost) if the user is not a wpcom user.
</blockquote>

We don't want to generate a new ID each time the app restarts because this doesn't match up with the way we have things on Mixpanel. We *do* want to track .org users between sessions so we can see how they use the app as compared to .com users.